internal/resources/smbshare.go

```diff
@@ -0,0 +1,217 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	sambaoperatorv1alpha1 "github.com/samba-in-kubernetes/samba-operator/api/v1alpha1"
+	"github.com/samba-in-kubernetes/samba-operator/internal/conf"
+)
+
+// SmbShareManager is used to manage SmbShare resources.
+type SmbShareManager struct {
+	client client.Client
+	scheme *runtime.Scheme
+	logger Logger
+}
+
+// NewSmbShareManager creates a SmbShareManager.
+func NewSmbShareManager(client client.Client, scheme *runtime.Scheme, logger Logger) *SmbShareManager {
+	return &SmbShareManager{
+		client: client,
+		scheme: scheme,
+		logger: logger,
+	}
+}
+
+// Update should be called when a SmbShare resource changes.
+func (m *SmbShareManager) Update(ctx context.Context, nsname types.NamespacedName) Result {
+	instance := &sambaoperatorv1alpha1.SmbShare{}
+	err := m.client.Get(ctx, nsname, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return Done
+		}
+		// Error reading the object - requeue the request.
+		return Done
```

Comment:
Not sure if this is what you intended, but comment and code don't seem to match.

```suggestion
		return Requeue
```

Reply:
You are right. This mistake is present in the other CR managers and I copy and pasted it here. I have a fix in a change for a future PR. If you like, we can fix it in this PR, or we can ignore it temporarily and take the fix in a follow up. Let me know how you feel.

Reply:
why not get it right when we add it?